Feature/waku v2 historical sketch

oskarth opened the pull request:
Addresses https://github.com/vacp2p/specs/issues/164

oskarth committed:
Waku v2: Add content filter for waku topics

This addresses https://github.com/vacp2p/specs/issues/156 and
https://github.com/vacp2p/specs/issues/160

oskarth commented:
Ping @decanus @kdeme

oskarth commented:
Also this spell check is annoying af

decanus commented on specs/waku/waku-v2.md:
```diff
@@ -115,11 +115,18 @@ In this section we specify two things:
 	repeated Message publish = 2;
+    repeated ContentFilter contentFilter = 3;
+    repeated HistoryQuery historyQuery = 4;
+    repeated HistoryResponse historyResponse = 5;
```
```suggestion
    repeated SubOpts subscriptions = 1;
    repeated Message publish = 2;
    repeated ContentFilter contentFilter = 3;
    repeated HistoryQuery historyQuery = 4;
    repeated HistoryResponse historyResponse = 5;
```

decanus commented on specs/waku/waku-v2.md:
```diff
@@ -115,11 +115,18 @@ In this section we specify two things:
 	message SubOpts {
 		optional bool subscribe = 1;
 		optional string topicid = 2;
 	}
```
```suggestion
    message SubOpts {
        optional bool subscribe = 1;
        optional string topicid = 2;
    }
```

decanus commented on specs/waku/waku-v2.md:
```diff
@@ -115,11 +115,18 @@ In this section we specify two things:
 	}
+
+    message ContentFilter {
+    optional string subtopic = 1;
```
```suggestion
        optional string subtopic = 1;
```

decanus commented on specs/waku/waku-v2.md:
```diff
@@ -129,14 +136,32 @@ message Message {
 	repeated string topicIDs = 4;
 	optional bytes signature = 5;
 	optional bytes key = 6;
+    optional string contentTopic = 7;
```
```suggestion
    repeated string topicIDs = 4;
    optional bytes signature = 5;
    optional bytes key = 6;
    optional string contentTopic = 7;
```

oskarth committed:
Fix spellcheck and indent

oskarth committed:
More protobuf fmt

oskarth committed:
Waku v2: Shape up historical API

Addresses https://github.com/vacp2p/specs/issues/164

Also spellcheck and protobuf formating

oskarth commented:
Same as https://github.com/vacp2p/specs/pull/163 comments @decanus

oskarth committed:
Spellcheck

oskarth commented:
@decanus can you please review/approve this.

decanus commented on specs/waku/waku-v2.md:
```diff
@@ -113,30 +113,54 @@ In this section we specify two things:
 
+### RPC
+
+These are messages sent to directly connected peers. These SHOULD NOT be
```
```suggestion
These are messages sent to directly connected peers. They SHOULD NOT be
```

oskarth committed:
Update specs/waku/waku-v2.md

Co-authored-by: Dean Eigenmann <7621705+decanus@users.noreply.github.com>

oskarth committed:
Merge branch 'master' into feature/waku-v2-historical-sketch

oskarth commented:
Should we turn off this double reviewing, kind of annoying
